overlord/ifacestate/ifacemgr.go

```diff
@@ -55,8 +61,10 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 
 	// Leave udevRetryTimeout at the default value, so that udev is initialized on first Ensure run.
 	m := &InterfaceManager{
-		state: s,
-		repo:  interfaces.NewRepository(),
+		state:                s,
+		repo:                 interfaces.NewRepository(),
+		enumeratedDeviceKeys: make(map[string]map[string]bool),
```

Comment:
this probably merits a comment that it is != nil so far device enumeration is done. I also wonder if a more explicit flag as well would be less subtle.

Reply:
Added a comment and a flag.